### ISSUE TITLE
Move keras.efficientnet preprocessing layers to preprocess_input function

### ISSUE THREAD

**Issue by JoshEZiegler** [tensorflow/tensorflow#44745]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.3.1
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
Current [preprocessing layers](https://github.com/tensorflow/tensorflow/blob/fcc4b966f1265f466e82617020af93670141b009/tensorflow/python/keras/applications/efficientnet.py#L316-L317) are a part of the main model. They are also not noted in the [TF docs](https://www.tensorflow.org/api_docs/python/tf/keras/applications/EfficientNetB0). 

I would propose moving these layers to the efficientnet preprocess_input function which currently does nothing. This would be consistent with the behavior of the other ready-made keras models.

**Will this change the current api? How?**
Yes. It would move preprocessing layers into a separate function. Unfortunately this would likely lead to unexpected behavior but not errors for people currently using the built-in preprocessing layers.

**Who will benefit with this feature?**
New users of the EfficientNet ready-made models. It would make the api consistent with other ready-made models. It would also separate preprocessing for edge case applications.

**Any Other info.**

**Comment by fchollet** [comment]
All Keras Applications going forward with do preprocessing as part of the model, so that they are able to take as input raw images (which removes the need to think about preprocessing at all). Old applications will keep using `preprocessing_input` for backwards compatibility.

**Comment by bayethiernodiop** [comment]
@fchollet but in the efficientnet original implementation, they use the preprocess in imagenet_utils with torch mode which have these fixed means and std for normalization
` x = layers.Rescaling(1. / 255.)(x)
  x = layers.Normalization(axis=bn_axis)(x)`
 but in the the peprocess in the efficientNet class are just:
` x = layers.Rescaling(1. / 255.)(x)
  x = layers.Normalization(axis=bn_axis)(x)`  
This normalization will compute mean and std depending on the input, is it possible that this may cause bad performance when the model is just used for extracting image features.

**Comment by JoshEZiegler** [comment]
@bayethiernodiop I was confused about the normalization layers in the loaded EfficientNet models as well. From the docs on [normalization layers](https://www.tensorflow.org/api_docs/python/tf/keras/layers/experimental/preprocessing/Normalization#examples_2) it looks like the layer has to be called with `layer.adapt` to change the mean and std, so I think the EfficientNet's normalization layers don't change with the input data. 

This seems to be case with my own experience. I was able to feed it unnormalized data with various scalings and with varying results, as you'd expect if the normalization layer was static.

**Comment by starrabb1t** [comment]
Can't convert to tfjs model because of preprocessing layers. It's not cool

**Comment by starrabb1t** [comment]
Guys, you can use [qubvel](https://github.com/qubvel/efficientnet) implementation. The only difference is the absence of Rescale and Normalization layers exactly!

**Comment by chrisrapson** [comment]
@fchollet can you please provide more information (or a link) on what will be different for "new" and "old" applications? And from what versions of keras and TF will the change occur? (Or has it already been released?) Will all pre-trained models from that point on expect input between [0,255], as opposed to previously expecting a range of [-1,1]? Will the expected data type still be float, or will that change to uint8?

I don't have a strong preference which range or datatype is chosen, the most important thing is to be consistent across all models. It was confusing for me why EfficientNet was behaving differently to other keras models. Since this is going to cause subtle breakages for many people, I hope it will be communicated clearly and well in advance.

See also issue #42506
